### ISSUE TITLE
Issue with ApiAuth.Login response

### ISSUE THREAD

**Issue by timresch** [ordercloud-api/OrderCloud-CSharp-SDK#1]
The ApiAuth.Login fails with the following exception messages:

Request to https://stagingauth.ordercloud.io/oauth/token/OAuth/Token failed. Unexpected character encountered while parsing value: T. Path '', line 0, position 0.

This occurs when a correct username/password combination is provided.
If an incorrect password is provided, the expected invalid logon error message is returned.

**Comment by cobrien451** [comment]
Looks like OAuth/Token is duplicated in the request URL. What did you set the base authorization URL to be? It should only be "https://stagingauth.ordercloud.io" to connect to staging. The other params are added for you by the SDK.

**Comment by timresch** [comment]
Yes, I caught that, and now I just get a generic “Exception of type 'Four51.ApiErrorException' was thrown."

Tim

**Comment by cobrien451** [comment]
[Test]
      public async Task CanLoginWithPasswordFlowTest()
      {
          var config = Configuration.Default;
          config.AuthURL = "https://stagingauth.ordercloud.io";
          config.ApiClient = new ApiClient("https://stagingapi.ordercloud.io/v1");
          config.ClientID = "***********-****-****-****-***********";
          config.Scopes = new[] {"FullAccess"};
          var auth = new AuthApi();
          var _me = new MeApi();
          var user = "CONNOR_OBRIEN";
          var password = "***********************";
          await auth.Login(user, password);
          var me = await _me.GetAsync();
          Assert.AreEqual(me.Username, user);
      }

Anything different between my test and your use case? Also responded to your email but wanted to leave this here as a reverence.

**Comment by timresch** [comment]
Here is my current code snippet – I have removed all variables and hard-coded the connection information:

using OrderCloudSDK.Api;
using OrderCloudSDK.Client;

…

            var config = Configuration.Default;
            config.ClientID = "35C0B8D9-4FAD-4544-902B-3AA1B8DC5630"; // weirDev
            config.AuthURL = "https://stagingauth.ordercloud.io";
            config.ApiClient = new ApiClient("https://stagingapi.ordercloud.io/v1");
            config.Scopes = new [] {"FullAccess"};

            Username = "TRTest";
            Password = "********";

            AuthApi auth = new AuthApi();
            await auth.Login(Username, Password);

I consistently get the “Exception of type 'Four51.ApiErrorException' was thrown” message (and it does not appear to matter whether I use an invalid password or the correct one)

Tim

**Comment by timresch** [comment]
Correction – it appears that a password change did not work. I am able to connect with the “old” password; an incorrect password gives the generic error message.

Tim

From: Resch, Tim
Sent: Monday, April 17, 2017 3:13 PM
To: 'ordercloud-api/OrderCloud-CSharp-SDK' <reply@reply.github.com>; ordercloud-api/OrderCloud-CSharp-SDK <OrderCloud-CSharp-SDK@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [ordercloud-api/OrderCloud-CSharp-SDK] Issue with ApiAuth.Login response (#1)

Here is my current code snippet – I have removed all variables and hard-coded the connection information:

using OrderCloudSDK.Api;
using OrderCloudSDK.Client;

…

            var config = Configuration.Default;
            config.ClientID = "35C0B8D9-4FAD-4544-902B-3AA1B8DC5630"; // weirDev
            config.AuthURL = "https://stagingauth.ordercloud.io";
            config.ApiClient = new ApiClient("https://stagingapi.ordercloud.io/v1");
            config.Scopes = new [] {"FullAccess"};

            Username = "TRTest";
            Password = "********";

            AuthApi auth = new AuthApi();
            await auth.Login(Username, Password);

I consistently get the “Exception of type 'Four51.ApiErrorException' was thrown” message (and it does not appear to matter whether I use an invalid password or the correct one)

Tim

**Comment by cobrien451** [comment]
I'll try and authenticate with an incorrect password locally, just need to verify if this is and SDK issue where it is only passing back a generic error message or if the API isn't sending the correct information.

**Comment by cobrien451** [comment]
Alright so I double checked with our QA lead and what you are experiencing is a known issue with our auth service in staging and we are currently working to correct our error messaging. I have added myself as a watcher to the issue and will try and inform you as soon as it has been resolved. Also maybe information you already know but do to refactoring around our password hashing algorithm all user passwords in staging will require a reset. You might be able to initially login in successfully but your token will only contain the PasswordReset role and will only allow that user access to the Me.Get, Me.ResetPasswordByToken routes.
